src/OrchardCore/OrchardCore.Abstractions/Shell/ShellSettings.cs

```diff
@@ -44,6 +45,12 @@ public string VersionId
             set => _settings["VersionId"] = value;
         }
 
+        public string TenantId
+        {
+            get => _settings["TenantId"];
+            set => _settings["TenantId"] = value;
```

Comment:
@sebastienros I would have preferred that this be an init set and used in the constructor that is used for the setup only. But that seems not possible to be done based on @jtkech comments on my initial PR.